lib/ansible/modules/network/aos/aos_login.py

```diff
@@ -0,0 +1,129 @@
+#
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: aos_login
+author: jeremy@apstra.com (@jeremyschulman)
+version_added: "2.3"
+short_description: Login to AOS server for session token
+description:
+  - Obtain the AOS server session token by providing the required
+    username and password credentials.  Upon successful authentication,
+    this module will return the session-token that is required by all
+    subsequent AOS module usage. On success the module will automatically populate
+    ansible facts with the variable I(aos_session)
+    This module is not idempotent and do not support chech mode.
+requirements:
+  - "aos-pyez >= 0.6.0"
+options:
+  server:
+    description:
+      - Address of the AOS Server on which you want to open a connection.
+    required: true
+  port:
+    description:
+      - Port number to use when connecting to the AOS server.
+    required: false
+  user:
+    description:
+      - Login username to use when connecting to the AOS server.
+    required: false
+    default: admin
+  passwd:
+    description:
+      - Password to use when connecting to the AOS server.
+    required: false
+    default: admin
+'''
+
+EXAMPLES = '''
+
+- name: Create a session with the AOS-server
+  aos_login:
+    server: "{{ inventory_hostname }}"
+    user: admin
+    passwd: admin
+
+- name: Use the newly created session (register is not mandatory)
+  aos_ip_pool:
+    session: "{{ aos_session }}"
+    name: my_ip_pool
+    state: present
+'''
+
+RETURNS = '''
+aos_session:
+  description: Authenticated session information
+  returned: always
+  type: dict
+  sample: { 'url': <str>, 'headers': {...} }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    from apstra.aosom.session import Session
+    import apstra.aosom.exc as aosExc
+
+    HAS_AOS_PYEZ = True
+except ImportError:
+    HAS_AOS_PYEZ = False
+
+def aos_login(module):
+
+    mod_args = module.params
+
+    aos = Session(server=mod_args['server'], port=mod_args['port'],
+                  user=mod_args['user'], passwd=mod_args['passwd'])
+
+    try:
+        aos.login()
+    except aosExc.LoginServerUnreachableError:
+        module.fail_json(
+            msg="AOS-server [%s] API not available/reachable, check server" % aos.server)
+
+    except aosExc.LoginAuthError:
+        module.fail_json(msg="AOS-server login credentials failed")
+
+    module.exit_json(changed=False,
+                     ansible_facts=dict( aos_session=dict(url=aos.api.url, headers=aos.api.headers)),
+                     aos_session=dict(url=aos.api.url, headers=aos.api.headers))
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server=dict(required=True),
+            port=dict(default='8888'),
+            user=dict(default='admin'),
+            passwd=dict(default='admin', no_log=True)))
```

Comment:
Just an FYI.
I believe the way `no_log` works is my replacing any occurrence of that string in output displayed to the users. Therefore  if `admin` is both username and password then we wouldn't see the username if it was returned in a message.